test/parallel/test-loader.js

```diff
@@ -0,0 +1,56 @@
+import { tryPackage }  from '../../lib/internal/modules/cjs/loader'
```

Comment:
This can be tested using the public APIs alone, so I would suggest you to revert the `tryPackage` exporting change from `lib/internal/modules/cjs/loader.js` and instead create a temporary file structure of this sort dynamically while running the test:
```js
❯ tree -a
.
├── index.js
└── node_modules
    ├── a
    │   ├── a.js
    │   └── package.json
    └── b
        └── package.json

3 directories, 4 files
❯ cat index.js
require('a');
❯ cat node_modules/a/a.js
require('b');
❯ cat node_modules/a/package.json
{
  "main": "a.js"
}
❯ cat node_modules/b/package.json
{
  "main": "b.js"
}
```
and assert on the `requireStack` property of the error thrown by the `index.js` file.

Also, note that this is a CJS file, so the way you used import here won't work.